Sources/ProjectDescription/RunActionOptions.swift

```diff
@@ -27,8 +35,113 @@ public struct RunActionOptions: Equatable, Codable {
+        )
+    }
+}
+
+extension RunActionOptions {
+    /// Represents a simulated location used when running the provided run action
+    public enum SimulatedLocation {
+        case london
+        case johannesburg
+        case moscow
+        case mumbai
+        case tokyo
+        case sydney
+        case hongKong
+        case honolulu
+        case sanFrancisco
+        case mexicoCity
+        case newYork
+        case rioDeJaneiro
+        case custom(gpxFile: Path)
+
+        /// A unique identifier string for the selected simulated location.
+        ///
+        /// In case of Xcode's simulated locations, this is a string representing the location.
+        /// In case of a custom GPX file, this is a path to that file.
+        public var identifier: String {
+            switch self {
+            case .london:
+                return "London, England"
+            case .johannesburg:
+                return "Johannesburg, South Africa"
+            case .moscow:
+                return "Moscow, Russia"
+            case .mumbai:
+                return "Mumbai, India"
+            case .tokyo:
+                return "Tokyo, Japan"
+            case .sydney:
+                return "Sydney, Australia"
+            case .hongKong:
+                return "Hong Kong, China"
+            case .honolulu:
+                return "Honolulu, HI, USA"
+            case .sanFrancisco:
+                return "San Francisco, CA, USA"
+            case .mexicoCity:
+                return "Mexico City, Mexico"
+            case .newYork:
+                return "New York, NY, USA"
+            case .rioDeJaneiro:
+                return "Rio De Janeiro, Brazil"
+            case let .custom(path):
+                return path.pathString
+            }
+        }
+    }
+}
+
+extension RunActionOptions.SimulatedLocation: Equatable, Codable {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let value = try container.decode(String.self)
+
+        switch value {
+        case "London, England":
+            self = .london
+        case "Johannesburg, South Africa":
+            self = .johannesburg
+        case "Moscow, Russia":
+            self = .moscow
+        case "Mumbai, India":
+            self = .mumbai
+        case "Tokyo, Japan":
+            self = .tokyo
+        case "Sydney, Australia":
+            self = .sydney
+        case "Hong Kong, China":
+            self = .hongKong
+        case "Honolulu, HI, USA":
+            self = .honolulu
+        case "San Francisco, CA, USA":
+            self = .sanFrancisco
+        case "Mexico City, Mexico":
+            self = .mexicoCity
+        case "New York, NY, USA":
+            self = .newYork
+        case "Rio De Janeiro, Brazil":
+            self = .rioDeJaneiro
+        case _ where value.contains("/"):
+            self = .custom(gpxFile: Path(value))
```

Comment:
When using relative paths there may not always be a `/` in the final value:

e.g.

```swift
let userScheme = Scheme(name: "App-Local",
                        shared: false,
                        buildAction: BuildAction(targets: ["App"], preActions: [debugAction]),
                        testAction: TestAction(targets: ["AppTests"]),
                        runAction: RunAction(
                            executable: "App",
                            options: .options(
                                simulatedLocation: .custom(gpxFile: "MyGpxFile.gpx")
                            )
                        ))
```

This will yield an `CodingError.unknownLocation` coding error, encoding the type explicitly may help (a bit of hassle, but can remove ambiguity).

A unit test for the custom codable may be worthwhile, `RunActionOptions` doesn't have one yet, but you could copy any of the existing `ProjectDescriptionTests` as a starter for it.

An alternative to ease the codable dance is to cheat slightly and use a struct with optionals however expose the desired cases as memberwise static initialisers e.g.

```swift
public struct SimulatedLocation: Codable {
    public let identifier: String?
    public let gpxFile: Path?

    public static var london: SimulatedLocation {
        SimulatedLocation(identifier: "London, England", gpxFile: nil)
    }

    // ...

    public static func custom(gpxFile: Path) -> SimulatedLocation {
        SimulatedLocation(identifier: nil, gpxFile: gpxFile)
    }
}
```

The only so-so compromise is we have to assume we'll never have a case where both variables are defined or empty as we're not exposing the direct initializer (enums are supposed to be the solution to this, but as soon as you add associated types, you need to pay the codable tax!)